lib/puppet/file_system.rb

```diff
@@ -212,6 +212,12 @@ def self.mkpath(path)
     @impl.mkpath(assert_path(path))
   end
 
+  def self.absolute_path?(path)
+    return false if path.nil?
```

Comment:
Should really find all the current callers passing `nil` - their logic is likely flawed.